server/polar/organization/schemas.py

```diff
@@ -86,34 +86,31 @@ class OrganizationCreate(OrganizationPrivateBase):
     pledge_minimum_amount: int = settings.MINIMUM_ORG_PLEDGE_AMOUNT
 
     @classmethod
-    def from_github_installation(
-        cls, installation: github.rest.Installation
-    ) -> OrganizationCreate:
-        account = installation.account
-
-        if not isinstance(account, github.rest.SimpleUser):
-            raise Exception("Polar does not support GitHub Enterprise")
-
-        is_personal = account.type.lower() == "user"
-        name = account.login
-        avatar_url = account.avatar_url
-        external_id = account.id
-        if not name:
-            raise Exception("repository.name is not set")
-        if not avatar_url:
-            raise Exception("repository.avatar_url is not set")
-
-        return cls(
-            platform=Platforms.github,
-            name=name,
-            external_id=external_id,
-            avatar_url=avatar_url,
-            is_personal=is_personal,
-            installation_id=installation.id,
-            installation_created_at=installation.created_at,
-            installation_updated_at=installation.updated_at,
-            installation_suspended_at=installation.suspended_at,
-        )
+    def from_github(
+        cls,
+        user: github.webhooks.User | github.rest.SimpleUser,
+        *,
+        installation: github.rest.Installation
+        | github.webhooks.Installation
+        | None = None,
+    ) -> Self:
+        data: dict[str, Any] = {
+            "is_personal": user.type.lower() == "user",
+            "name": user.login,
+            "avatar_url": user.avatar_url,
+            "external_id": user.id,
+        }
+
+        if installation is not None:
+            data = {
+                **data,
+                "installation_id": installation.id,
+                "installation_created_at": installation.created_at,
+                "installation_updated_at": installation.updated_at,
+                "installation_suspended_at": installation.suspended_at,
+            }
+
+        return cls(platform=Platforms.github, **data)
```

Comment:
🤔 Instead of adding `Any` and `**` can we split this into two classes? One with installation data and one without. That way we can still be type-safe and maybe even remove `exclude_unset`?

Reply:
I kinda like to have a single method that can handle both, but definitely doable to do so with more type-safety.

The `exclude_unset` trick is actually not needed given how we structured our schema. I can remove this.